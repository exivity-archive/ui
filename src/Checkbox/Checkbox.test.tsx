import { mount, shallow } from 'enzyme'
import * as React from 'react'

import { Checkbox } from '.'

test('renders checkbox', () => {
  const checkbox = shallow(<Checkbox checked onClick={jest.fn()} />)
  expect(checkbox).toMatchSnapshot()
})

test('onClick handler returns checkedValue', () => {
  const onClickMock = jest.fn(x => x)

  const checkbox = mount(<Checkbox checked onClick={onClickMock} />)

  checkbox
    .find({ type: 'checkbox' })
    .simulate('click', { target: { checked: false } })

  expect(onClickMock.mock.results[0].value).toBe(false)
})

test('onChange handler returns checkedValue', () => {
  const onChangeMock = jest.fn(x => x)

  const checkbox = mount(<Checkbox checked onChange={onChangeMock} />)

  checkbox
    .find({ type: 'checkbox' })
    .simulate('change', { target: { checked: false } })

  expect(onChangeMock.mock.results[0].value).toBe(false)
})

test('checked prop to true', () => {
  const checkbox = shallow(<Checkbox checked onChange={jest.fn()} />)
  expect(checkbox.props().checked).toBe(true)
})

test('checked prop to false', () => {
  const checkbox = shallow(<Checkbox checked={false} onChange={jest.fn()} />)
  expect(checkbox.props().checked).toBe(false)
})

test('checked prop updates', () => {
  const checkbox = mount(<Checkbox checked={false} onChange={jest.fn()} />)

  const checked = checkbox
    .setProps({ checked: true })
    .find({ type: 'checkbox' })
    .props()
    .checked

  expect(checked).toBe(true)
})
