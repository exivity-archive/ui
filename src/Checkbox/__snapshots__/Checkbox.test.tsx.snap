// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders checkbox 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForwardRef
    checked={true}
    onClick={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "checked": true,
      "forwardedComponent": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": false,
          "rules": Array [
            "
  width: 20px;
  height: 20px;
  cursor: pointer;
  background: ",
            [Function],
            ";
  color: black;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
  position: relative;
  left: -5px;
  top: -5px;

  :focus {
    outline:0;
  }
",
          ],
        },
        "displayName": "Styled(PlainCheckbox)",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      },
      "forwardedRef": null,
      "onClick": [MockFunction],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "checked": true,
        "forwardedComponent": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "rules": Array [
              "
  width: 20px;
  height: 20px;
  cursor: pointer;
  background: ",
              [Function],
              ";
  color: black;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
  position: relative;
  left: -5px;
  top: -5px;

  :focus {
    outline:0;
  }
",
            ],
          },
          "displayName": "Styled(PlainCheckbox)",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
        "forwardedRef": null,
        "onClick": [MockFunction],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
